pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -105,6 +105,48 @@ type Reconciler struct {
 	GetMemberClusters cluster.GetMemberClustersFunc
 }
 
+/* =================================================
+   Migration code START - remove after migration complete
+  =================================================*/
+
+func migrateApproved(userSignup *toolchainv1alpha1.UserSignup) {
+	if userSignup.Spec.Approved {
+		setState(userSignup, toolchainv1alpha1.UserSignupStateApproved, true)
+	}
+}
+
+func setState(userSignup *toolchainv1alpha1.UserSignup, state toolchainv1alpha1.UserSignupState, val bool) {
+	if val && !contains(userSignup.Spec.States, state) {
+		userSignup.Spec.States = append(userSignup.Spec.States, state)
+	}
+
+	if !val && contains(userSignup.Spec.States, state) {
+		userSignup.Spec.States = remove(userSignup.Spec.States, state)
+	}
+}
+
+func contains(s []toolchainv1alpha1.UserSignupState, state toolchainv1alpha1.UserSignupState) bool {
+	for _, a := range s {
+		if a == state {
+			return true
+		}
+	}
+	return false
+}
+
+func remove(s []toolchainv1alpha1.UserSignupState, state toolchainv1alpha1.UserSignupState) []toolchainv1alpha1.UserSignupState {
+	for i, v := range s {
+		if v == state {
+			return append(s[:i], s[i+1:]...)
+		}
+	}
+	return s
+}
```

Comment:
I'm confused? Why not using the state manager https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/states/state_manager.go#L58-L84

Reply:
We don't use the state manager here because using it to set Approved will cause the Deactivating and Deactivated states to be set to false.  This may result in already-deactivated users becoming reactivated again.

Reply:
I thought that this was solved by: 
https://github.com/sbryzak/host-operator/blob/38ac9b78b3876b0b9e58ab54ea8ed6c1dc626ad3/pkg/controller/usersignup/usersignup_controller.go#L176